com.unity.multiplayer.mlapi/Runtime/Metrics/NetworkMetrics.cs

```diff
@@ -38,24 +52,42 @@ public class NetworkMetrics : INetworkMetrics
         private EventMetric<NamedMessageEvent> m_NamedMessageSentEvent = new EventMetric<NamedMessageEvent>("Named Message Sent");
         private EventMetric<NamedMessageEvent> m_NamedMessageReceivedEvent = new EventMetric<NamedMessageEvent>("Named Message Received");
 
+        private Dictionary<ulong, NetworkObjectIdentifier> m_NetworkGameObjects = new Dictionary<ulong, NetworkObjectIdentifier>();
+
         public NetworkMetrics(NetworkManager networkManager)
         {
             m_NetworkManager = networkManager;
             m_Dispatcher = new MetricDispatcherBuilder()
                 .WithMetricEvents(m_NamedMessageSentEvent, m_NamedMessageReceivedEvent)
                 .Build();
-            
+
             m_Dispatcher.RegisterObserver(MLAPIObserver.Observer);
         }
 
-        public void TrackNamedMessageSent(string messageName, ulong bytesCount)
+        public void TrackNetworkObject(NetworkObject networkObject)
+        {
+            if (!m_NetworkGameObjects.ContainsKey(networkObject.NetworkObjectId))
+            {
+                m_NetworkGameObjects[networkObject.NetworkObjectId] = new NetworkObjectIdentifier(networkObject.name, networkObject.NetworkObjectId);
+            }
+        }
+
+        public void TrackNamedMessageSent(ulong clientId, string messageName, ulong bytesCount)
```

Comment:
I think we should be very explicit about client IDs and really call them "sender" and "receiver" whenever we use them.

Reply:
I'll update this and merge.